### PR TITLE
Remove polyfills file

### LINKED ISSUE #14432: Jasmine's clock mocks can't be used in library specs
# 🐞 Bug report

### Command (mark with an `x`)
```
- [ ] new
- [ ] build
- [ ] serve
- [x] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?
Not sure to be honest, but it works if the sample spec is part of the application in the workspace.

### Description
Attempting to use Jasmine's clock mocks inside tests of CLI generated libraries throws an error saying 'Jasmine Clock was unable to install over custom global timer functions.'. It seems Jasmine is imported before Zone.js and therefore detects that the global timing functions, such as setTimeout, were changed afterwoods.

The issue is similar to https://github.com/angular/angular-cli/issues/11626.

## 🔬 Minimal Reproduction
<pre><code>
1. npm install --global @angular/cli.
2. ng new test-app
3. cd test-app
4. ng g library time-lib
5. ng test time-lib (passes)

6. Add the following test to a spec file inside the lib-1
 (e.g.: into "time-lib.service.spec.ts"):
 
 it('should install the clock mock', () => {
    jasmine.clock().install();
    jasmine.clock().uninstall();
  });

7. ng test time-lib (fails)
</code></pre>
## 🔥 Exception or Error
<pre><code>
Chrome 74.0.3729 (Windows 10.0.0) TimeLibService should install the clock mock FAILED
        Error: Jasmine Clock was unable to install over custom global timer functions. Is the clock already installed?
            at UserContext.<anonymous> (src/lib/time-lib.service.spec.ts:14:21)
            at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (C:/DEV/_test_projects/test-app/node_modules/zone.js/dist/zone.js:391:1)
            at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (C:/DEV/_test_projects/test-app/node_modules/zone.js/dist/zone-testing.js:289:1)
            at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (C:/DEV/_test_projects/test-app/node_modules/zone.js/dist/zone.js:390:1)
            at Zone.push.../../node_modules/zone.js/dist/zone.js.Zone.run (C:/DEV/_test_projects/test-app/node_modules/zone.js/dist/zone.js:150:1)
            at runInTestZone (C:/DEV/_test_projects/test-app/node_modules/zone.js/dist/zone-testing.js:510:1)
            at UserContext.<anonymous> (C:/DEV/_test_projects/test-app/node_modules/zone.js/dist/zone-testing.js:525:1)
</code></pre>

## 🌍 Your Environment
<pre><code>
Angular CLI: 7.3.9
Node: 8.11.2
OS: win32 x64
Angular: 7.2.15
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.13.9
@angular-devkit/build-angular      0.13.9
@angular-devkit/build-ng-packagr   0.13.9
@angular-devkit/build-optimizer    0.13.9
@angular-devkit/build-webpack      0.13.9
@angular-devkit/core               7.3.9
@angular-devkit/schematics         7.3.9
@angular/cli                       7.3.9
@ngtools/json-schema               1.1.0
@ngtools/webpack                   7.3.9
@schematics/angular                7.3.9
@schematics/update                 0.13.9
ng-packagr                         4.7.1
rxjs                               6.3.3
typescript                         3.2.4
webpack                            4.29.0
</code></pre>

**Anything else relevant?**
Based on the fix for https://github.com/angular/angular-cli/issues/11626, adding a (e.g.) <code>test-pollyfill.ts</code> to the library source and moving the zone.js import from the library's <code>test.ts</code> file into the test-pollyfill.ts file, then configuring the test architecture of the library's project in <code>angular.json</code>to use the file as the 'pollyFills' option seems to be a possible workaround for now.

### PR DESCRIPTION
**Pull request by alan-agius4**
**feat(@angular-devkit/build-angular): amend `polyfills` option in all builders to support an array of module specifiers**
    
This is the ground work to be able to remove the `polyfills.ts` file which today is primarily used to add `zone.js`.
    
Usage examples:
```js
polyfills: ['zone.js'],

polyfills: ['zone.js', 'zone.js/testing', 'src/polyfills.ts'],
```

**feat(@schematics/angular): drop `polyfills.ts` file from new templates**
    
With this change we drop the `polyfills.ts` from new application projects and add the polyfills directly in the `angular.json`. This is possible as now the `polyfills` option accept an array of module specifiers.
    
This change also fixes another open issue (#14432) which was caused by the missing polyfills file in the library test setup.
    
Closes #14432

### CODE REVIEW

**Comment by e-oz** [comment]
Will it still be possible to define Zone.JS flags before importing Zone.JS?

**Comment by e-oz** [comment]
> You can still provide those in a polyfills.ts file.

If I remember correctly, the order of imports matters here. Flags should be defined before zone.js import.

**Comment by alan-agius4** [comment]
You can order the files to imports as you'd needed.

Example:
```
polyfills: ['./zone-flags.ts', 'zone.js'],
polyfills: ['./zone-flags.ts', 'zone.js', './other-polyfills.ts', ....],
```

**Comment by e-oz** [comment]
Thanks a lot for answering :)

File: packages/angular_devkit/build_angular/src/builders/browser-esbuild/options.ts [hunk 1]
```diff
@@ -33,10 +34,21 @@ export async function normalizeOptions(
     workspaceRoot,
     (projectMetadata.sourceRoot as string | undefined) ?? 'src',
   );
-
   // Normalize options
   const mainEntryPoint = path.join(workspaceRoot, options.main);
-  const polyfillsEntryPoint = options.polyfills && path.join(workspaceRoot, options.polyfills);
+
+  // Currently esbuild do not support multiple files per entry-point
+  const [polyfillsEntryPoint, ...remainingPolyfills] = normalizePolyfills(
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/builders/browser-esbuild/options.ts:41]: Can the path resolving step be skipped here?  The absolute paths will end up in the sourcemaps. 
esbuild should be able to resolve the relative paths directly.

**Review by clydin — CHANGES_REQUESTED** [review]

File: packages/angular_devkit/build_angular/src/builders/browser-esbuild/options.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/builders/browser-esbuild/options.ts:41]: As per off-line convo, lets do this in a follow up

**Review by clydin — APPROVED** [review]

**Pull request merged**